core/src/util/mod.rs

```diff
@@ -102,63 +102,92 @@ macro_rules! set {
 
 #[cfg(test)]
 mod filtering {
-    use super::{is_match, to_name_regex};
-
-    #[test]
-    fn test_to_name_regex() {
-        assert!(to_name_regex("%").is_none());
-        assert!(to_name_regex("").is_none());
-        assert!(to_name_regex("filter").is_some());
+    use super::{is_match, table_type_filter_to_vec, to_name_regex};
+
+    mod table_type_filter_to_vec {
+        use super::table_type_filter_to_vec;
+        use mongodb::results::CollectionType;
+        #[test]
+        fn test_table_type_filter_to_vec() {
+            assert_eq!(
+                table_type_filter_to_vec("table"),
+                Some(vec![CollectionType::Collection])
+            );
+            assert_eq!(
+                table_type_filter_to_vec("view"),
+                Some(vec![CollectionType::View])
+            );
+            assert_eq!(
+                table_type_filter_to_vec("table,view"),
+                Some(vec![CollectionType::Collection, CollectionType::View])
+            );
+            assert_eq!(table_type_filter_to_vec(""), None);
+            assert_eq!(table_type_filter_to_vec("%"), None);
+        }
     }
 
-    #[test]
-    fn test_is_positive_match_literal() {
-        assert!(is_match("%", "%", false));
-        assert!(is_match("%test", "%test", false));
-        assert!(is_match("down_times", "down_times", false));
-        assert!(is_match("filter", "filter", false));
-        assert!(is_match("downtimes", "downtimes", false));
-        assert!(is_match("money$$bags", "money$$bags", false));
-        assert!(is_match("money$.bags", "money$.bags", false));
+    mod to_name_regex {
+        use super::to_name_regex;
+        #[test]
+        fn test_to_name_regex() {
+            assert!(to_name_regex("%").is_none());
+            assert!(to_name_regex("").is_none());
+            assert!(to_name_regex("filter").is_some());
+        }
     }
 
-    #[test]
-    fn test_is_negative_match_literal() {
-        assert!(!is_match("filter", "%", false));
-        assert!(!is_match("customerssales", "customer_sales", false));
-        assert!(!is_match("conversions2022", "conversions%", false));
-        assert!(!is_match("integration_test", "%test", false));
-        assert!(!is_match("integration_test", "integrationstest", false));
-    }
+    mod is_match {
+        use super::is_match;
+
+        #[test]
+        fn test_is_positive_match_literal() {
```

Comment:
huh, usually functions like this put pattern before value to search, but I guess it's too late to change this :joy: